force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -81,15 +86,27 @@ public with sharing class ClassesObjectsMethods {
      */
     public static String removeDuplicateWords(String sentence) {
         // Split the sentence into words
-
         // Create a set to store the unique words
-
         // Add each word to the set
-
+        Set<String> words = new Set<String>();
+        while (String.isNotBlank(sentence) == true){
+            String word = sentence.substringBefore(' ');
+            words.add(word);
+            sentence = sentence.remove(word).trim();
+        }
         // Join the unique words back into a sentence
-
+        List<String> wordsList = new List<String>(words);
+        while(wordsList.size() > 0){
+            if(String.isEmpty(sentence)){
+                sentence = wordsList.get(0);
+            }
+            else{
+                sentence = sentence + ' ' + wordsList.get(0);
+            }
+        wordsList.remove(0);
+       }
         // Return the sentence with unique words
```

Comment:
This is a clever solution. And the first student I have seen do it with a while loop.  You can simplify this a lot. A Set can can accept a List into its constructor and there is no need for iteration at all.